nixos/gnome/gcr-ssh-agent: init (#379731)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2511.section.md

- `amdgpu` kernel driver overdrive mode can now be enabled by setting [hardware.amdgpu.overdrive.enable](#opt-hardware.amdgpu.overdrive.enable) and customized through [hardware.amdgpu.overdrive.ppfeaturemask](#opt-hardware.amdgpu.overdrive.ppfeaturemask).
  This allows for fine-grained control over the GPU's performance and maybe required by overclocking softwares like Corectrl and Lact. These new options replace old options such as {option}`programs.corectrl.gpuOverclock.enable` and {option}`programs.tuxclocker.enableAMD`.

> - {option}`services.ssh.startAgent`, who was previously using `gnome-keyring`,has been replaced with `gcr_4` due to being deprecated upstream. The option will be equally deprecated in the future and has been replaced with {option}`services.gnome.gcr-ssh-agent.enable`.

Reviewer: 1. Should we link to the option?
2. Grammar and phrasing could be improved 

My suggestion could still be improved on, but I think something like this is clearer:

```suggestion
- [{option}`services.ssh.startAgent`](#opt-services.ssh.startAgent), has switched from using `gnome-keyring`, to `gcr_4`. This is due to ssh-agent support being deprecated in the upstream `gnome-keyring` project.
  - {option}`services.ssh.startAgent` will itself be deprecated soon. The new [{option}`services.gnome.gcr-ssh-agent`](#opt-services.gnome.gcr-ssh-agent.enable) options should be used instead.
```

Ideally, we would also link to whether upstream announced the deprecation, assuming there's a sensible place to link to. (E.g. [gcr:67](https://gitlab.gnome.org/GNOME/gcr/-/merge_requests/67) is probably too technical to be worth linking here).

Is it also worth linking to https://github.com/NixOS/nixpkgs/issues/140824 ?

- `amdgpu` kernel driver overdrive mode can now be enabled by setting [hardware.amdgpu.overdrive.enable](#opt-hardware.amdgpu.overdrive.enable) and customized through [hardware.amdgpu.overdrive.ppfeaturemask](#opt-hardware.amdgpu.overdrive.ppfeaturemask).
  This allows for fine-grained control over the GPU's performance and maybe required by overclocking softwares like Corectrl and Lact. These new options replace old options such as {option}`programs.corectrl.gpuOverclock.enable` and {option}`programs.tuxclocker.enableAMD`.

- [{option}`programs.ssh.startAgent`](#opt-programs.ssh.startAgent), has switched from using `gnome-keyring`, to `gcr_4`. This is due to ssh-agent support being deprecated in the upstream `gnome-keyring` project. See the [relevant upstream PR](https://gitlab.gnome.org/GNOME/gcr/-/merge_requests/67) for more details.

>   - {option}`services.ssh.startAgent` will itself be deprecated soon. The new [{option}`services.gnome.gcr-ssh-agent`](#opt-services.gnome.gcr-ssh-agent.enable) options should be used instead.

Reviewer: :+1: awesome

>   - {option}`services.ssh.startAgent` will itself be deprecated soon. The new [{option}`services.gnome.gcr-ssh-agent`](#opt-services.gnome.gcr-ssh-agent.enable) options should be used instead.

Reviewer 2: ```suggestion
  - {option}`services.ssh.startAgent` is now deprecated and the new [{option}`services.gnome.gcr-ssh-agent`](#opt-services.gnome.gcr-ssh-agent.enable) option should be used instead in its place.
```

"will be soon" is ambiguous, with merge of this PR, it will be deprecated effective immediately. 

In my opinion, it would be better to have `ssh.startAgent` as an alias of the new module.

Since, NixOS 25.05 just got released, there is enough migration period/enough time for testing. And this isn't really a major change for us to wait that long.

Reviewer 3: By the way, the proper way to reference options in NixOS manual is just `[](#opt-services.gnome.gcr-ssh-agent.enable)`. The text of the link will be filled in automatically.
